sapi/fpm/fpm/fpm_stdio.c

```diff
@@ -144,12 +144,24 @@ static void fpm_stdio_child_said(struct fpm_event_s *ev, short which, void *arg)
 		zlog_stream_init_ex(log_stream, ZLOG_WARNING, STDERR_FILENO);
 		zlog_stream_set_decorating(log_stream, child->wp->config->decorate_workers_output);
 		zlog_stream_set_wrapping(log_stream, ZLOG_TRUE);
-		zlog_stream_set_msg_prefix(log_stream, "[pool %s] child %d said into %s: ",
-				child->wp->config->name, (int) child->pid, is_stdout ? "stdout" : "stderr");
 		zlog_stream_set_msg_quoting(log_stream, ZLOG_TRUE);
 	} else {
 		log_stream = child->log_stream;
+		// if fd type (stdout/stderr) is changed, then the stream will be finished
+		if (log_stream->is_stdout != (unsigned int)is_stdout) {
+			zlog_stream_finish(log_stream);
+		}
+		// free stream's msg_prefix for reset
+		if (log_stream->msg_prefix) {
+			free(log_stream->msg_prefix);
+			log_stream->msg_prefix = NULL;
+		}
 	}
+	// reset stream's msg_prefix
+	zlog_stream_set_msg_prefix(log_stream, "[pool %s] child %d said into %s: ",
+			child->wp->config->name, (int) child->pid, is_stdout ? "stdout" : "stderr");
+	// store fd type (stdout/stderr)
+	log_stream->is_stdout = (unsigned int)is_stdout;
```

Comment:
It changes only if they are different. It should be also wrapped in inline function as other flags.